Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1410,6 +1417,9 @@ namespace Babylon
 
         if (m_boundVertexArray != nullptr)
         {
+            if (!m_boundVertexArray->IsValid())
+                return;
```

Comment:
It seems like you are already storing the valid boolean inside vertex array and thus you can check inside the SetIndexBuffer and SetVertexBuffers function instead?